src/BuoyancyModels/nonlinear_equation_of_state.jl

```diff
@@ -7,15 +7,15 @@ const f = Face()
 
 """ Return the geopotential height at `i, j, k` at cell centers. """
 @inline Zᶜᶜᶜ(i, j, k, grid) =
-    ifelse(k < 1,       znode(i, j,       1, grid, c, c, c) + (1 - k) * Δzᶜᶜᶠ(i, j, 1, grid),
-    ifelse(k > grid.Nz, znode(i, j, grid.Nz, grid, c, c, c) - (k - grid.Nz) * Δzᶜᶜᶠ(i, j, grid.Nz, grid),
-                        znode(i, j,       k, grid, c, c, c)))
+    ifelse(k < 1,       znode(i, j,       1, grid, c, c, c, with_halos=true) + (1 - k) * Δzᶜᶜᶠ(i, j, 1, grid),
+    ifelse(k > grid.Nz, znode(i, j, grid.Nz, grid, c, c, c, with_halos=true) - (k - grid.Nz) * Δzᶜᶜᶠ(i, j, grid.Nz, grid),
+                        znode(i, j,       k, grid, c, c, c, with_halos=true)))
```

Comment:
I don't think we want a `with_halos` option for the singleton versions (ie functions of `i, j, k`). What would this mean anyways?

Reply:
I introduced those kwargs based on our discussion [here](https://github.com/CliMA/Oceananigans.jl/pull/2842/#discussion_r1065130759). I think that was a nice idea and it makes for fewer (and imho clearer) lines of code. I didn't know there was a downside to using them for GPUs.